internal/pkg/util/fs/layout/manager.go

```diff
@@ -389,7 +389,9 @@ func (m *Manager) sync() error {
 			}
 			if err := m.VFS.WriteFile(path, entry.content, entry.mode); err != nil {
 				if !os.IsExist(err) {
-					return fmt.Errorf("failed to create file %s: %s", path, err)
+					if _, err1 := os.Stat(path); err1 != nil {
```

Comment:
The approach makes sense, however like @DrDaveD suggested, it would be good to check if the path is a regular file when `err1 == nil`, because it could also be anything except a regular file, and in this case the bind mount will fail later, so better to check that here and return an appropriate error when it's the case

Reply:
There's not problem bind mounting on non-regular file. I tried to bind mount a regular file on a socket file and a named pipe and had no issue, it successfully mounted my regular file on both of them.

Reply:
Might worth to check if the existing path is not a directory at least because mount will fail in this case

Reply:
If the destination is a directory, opening it will fail with `EEXIST` anyway and won't even go into the `if !os.IsExist(err)` condition. By the way, apptainer will also fail in the mount stage also if the source is a directory and the destination is a file. There's nothing anywhere at the code that checks if the type of the source and the mountpoint match anyway.

Reply:
Ok thanks for the clarifications

Reply:
I would like to at least see a comment explaining what's going on here.